Index: admin/atracking.php
--- admin/atracking.php	2011-01-10 16:11:44.000000000 +0300
+++ admin/atracking.php	2011-04-28 14:52:14.000000000 +0400
@@ -40,20 +40,21 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: atracking.php,v 1.48.2.1 2011/01/10 13:11:44 ferz Exp $
+ * @version    $Id: atracking.php,v 1.48.2.4 2011/04/28 10:52:14 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
 
 if ( !defined('XCART_SESSION_START') ) { header("Location: ../"); die("Access denied"); }
 
-$date = "date+'".$config["Appearance"]["timezone_offset"]."'";
-$date_condition = "($date>='$start_date' AND $date<='$end_date')";
+$start_date_off = $start_date - $config["Appearance"]["timezone_offset"];
+$end_date_off = $end_date - $config["Appearance"]["timezone_offset"];
+$date_condition = "(date>='$start_date_off' AND date<='$end_date_off')";
 
-$cart_funnel = func_query_first_cell("SELECT COUNT(*) FROM $sql_tbl[stats_cart_funnel] WHERE $date_condition");
+$cart_funnel = func_query_first_cell("SELECT COUNT(date) FROM $sql_tbl[stats_cart_funnel] WHERE $date_condition");
 $product_funnel = func_query_first_cell("SELECT COUNT(pv.pageid) FROM $sql_tbl[stats_pages_views] as pv, $sql_tbl[stats_pages] as p WHERE p.pageid=pv.pageid AND p.page LIKE '%product.php?productid=%' AND $date_condition");
 
-$all_views = func_query_first_cell("SELECT COUNT(*) FROM $sql_tbl[stats_pages_views] WHERE $date_condition");
+$all_views = func_query_first_cell("SELECT COUNT(date) FROM $sql_tbl[stats_pages_views] WHERE $date_condition");
 
 if ($mode == 'pagesviews') {
 /**
@@ -75,7 +76,7 @@
  */
     $location[] = array(func_get_langvar_by_name('lbl_top_paths_thru_site'), '');
 
-    $all_views = func_query_first_cell("SELECT COUNT(*) FROM $sql_tbl[stats_pages_paths] WHERE path LIKE '%-%' AND $date_condition");
+    $all_views = func_query_first_cell("SELECT COUNT(date) FROM $sql_tbl[stats_pages_paths] WHERE path LIKE '%-%' AND $date_condition");
     $statistics = func_query("SELECT path, COUNT(path) as counter FROM $sql_tbl[stats_pages_paths] WHERE path LIKE '%-%' AND $date_condition GROUP BY path ORDER BY counter DESC LIMIT 25");
     if (!empty($statistics))
         foreach($statistics as $k => $v) {
@@ -84,7 +85,7 @@
                 foreach($pages as $k1=>$v1) {
                     $query = "SELECT p.page, ROUND(AVG(pv.time_avg),2) as time_avg FROM $sql_tbl[stats_pages_views] as pv, $sql_tbl[stats_pages] as p WHERE p.pageid=pv.pageid AND pv.pageid='$v1' AND $date_condition GROUP BY p.page";
                     $tmp = func_query_first($query);
-                    $statistics[$k][pages][] = $tmp;
+                    $statistics[$k]['pages'][] = $tmp;
                 }
             }
             $statistics[$k]['percent'] = $v['counter'] / $all_views * 100;
@@ -96,30 +97,30 @@
  */
     $location[] = array(func_get_langvar_by_name('lbl_shopping_cart_conversion_funnel'), '');
 
-    $statistics[0][step] = 'product_page';
-    $statistics[0][visits] = $product_funnel;
-    $statistics[0][percent_all] = sprintf("%2.02f", ( $statistics[0][visits] / ($all_views?$all_views:1) ) * 100);
-    $statistics[1][step] = 'start_page';
-    $statistics[1][visits] = func_query_first_cell("SELECT count(start_page) FROM $sql_tbl[stats_cart_funnel] WHERE start_page>'0' AND $date_condition");
-    $statistics[1][percent_parent] = sprintf("%2.02f", ( $statistics[1][visits] / ($statistics[0][visits]?$statistics[0][visits]:1) ) * 100);
-    $statistics[1][percent_all] = sprintf("%2.02f", ( $statistics[1][visits] / ($all_views?$all_views:1) ) * 100);
-    $statistics[2][step] = 'step1';
-    $statistics[2][visits] = func_query_first_cell("SELECT count(step1) FROM $sql_tbl[stats_cart_funnel] WHERE step1>'0' AND $date_condition");
-    $statistics[2][percent_parent] = sprintf("%2.02f", ( $statistics[2][visits] / ($statistics[1][visits]?$statistics[1][visits]:1) ) * 100);
-    $statistics[2][percent_all] = sprintf("%2.02f", ( $statistics[2][visits] / ($all_views?$all_views:1) ) * 100);
-    $statistics[3][step] = 'step2';
-    $statistics[3][visits] = func_query_first_cell("SELECT count(step2) FROM $sql_tbl[stats_cart_funnel] WHERE step2>'0' AND $date_condition");
-    $statistics[3][percent_parent] = sprintf("%2.02f", ( $statistics[1][visits] + $statistics[2][visits] ) == 0 ? 0 : (( $statistics[3][visits] / ( $statistics[1][visits] + $statistics[2][visits] ) ) * 100));
-    $statistics[3][percent_all] = sprintf("%2.02f", ( $statistics[3][visits] / ($all_views?$all_views:1) ) * 100);
-    $statistics[3][percent_parent2] = sprintf("%2.02f", ( $statistics[3][visits] / ($statistics[0][visits]?$statistics[0][visits]:1) ) * 100);
-    $statistics[4][step] = 'step3';
-    $statistics[4][visits] = func_query_first_cell("SELECT count(step3) FROM $sql_tbl[stats_cart_funnel] WHERE step3>'0' AND $date_condition");
-    $statistics[4][percent_parent] = sprintf("%2.02f", ( $statistics[4][visits] / ($statistics[3][visits]?$statistics[3][visits]:1) ) * 100);
-    $statistics[4][percent_all] = sprintf("%2.02f", ( $statistics[4][visits] / ($all_views?$all_views:1) ) * 100);
-    $statistics[5][step] = 'final_page';
-    $statistics[5][visits] = func_query_first_cell("SELECT count(final_page) FROM $sql_tbl[stats_cart_funnel] WHERE final_page>'0' AND $date_condition");
-    $statistics[5][percent_parent] = sprintf("%2.02f", ( $statistics[5][visits] / ($statistics[4][visits]?$statistics[4][visits]:1) ) * 100);
-    $statistics[5][percent_all] = sprintf("%2.02f", ( $statistics[5][visits] / ($all_views?$all_views:1) ) * 100);
+    $statistics[0]['step'] = 'product_page';
+    $statistics[0]['visits'] = $product_funnel;
+    $statistics[0]['percent_all'] = sprintf("%2.02f", ( $statistics[0]['visits'] / ($all_views?$all_views:1) ) * 100);
+    $statistics[1]['step'] = 'start_page';
+    $statistics[1]['visits'] = func_query_first_cell("SELECT count(start_page) FROM $sql_tbl[stats_cart_funnel] WHERE start_page>'0' AND $date_condition");
+    $statistics[1]['percent_parent'] = sprintf("%2.02f", ( $statistics[1]['visits'] / ($statistics[0]['visits']?$statistics[0]['visits']:1) ) * 100);
+    $statistics[1]['percent_all'] = sprintf("%2.02f", ( $statistics[1]['visits'] / ($all_views?$all_views:1) ) * 100);
+    $statistics[2]['step'] = 'step1';
+    $statistics[2]['visits'] = func_query_first_cell("SELECT count(step1) FROM $sql_tbl[stats_cart_funnel] WHERE step1>0 AND $date_condition");
+    $statistics[2]['percent_parent'] = sprintf("%2.02f", ( $statistics[2]['visits'] / ($statistics[1]['visits']?$statistics[1]['visits']:1) ) * 100);
+    $statistics[2]['percent_all'] = sprintf("%2.02f", ( $statistics[2]['visits'] / ($all_views?$all_views:1) ) * 100);
+    $statistics[3]['step'] = 'step2';
+    $statistics[3]['visits'] = func_query_first_cell("SELECT count(step2) FROM $sql_tbl[stats_cart_funnel] WHERE step2>0 AND $date_condition");
+    $statistics[3]['percent_parent'] = sprintf("%2.02f", ( $statistics[1]['visits'] + $statistics[2]['visits'] ) == 0 ? 0 : (( $statistics[3]['visits'] / ( $statistics[1]['visits'] + $statistics[2]['visits'] ) ) * 100));
+    $statistics[3]['percent_all'] = sprintf("%2.02f", ( $statistics[3]['visits'] / ($all_views?$all_views:1) ) * 100);
+    $statistics[3]['percent_parent2'] = sprintf("%2.02f", ( $statistics[3]['visits'] / ($statistics[0]['visits']?$statistics[0]['visits']:1) ) * 100);
+    $statistics[4]['step'] = 'step3';
+    $statistics[4]['visits'] = func_query_first_cell("SELECT count(step3) FROM $sql_tbl[stats_cart_funnel] WHERE step3>0 AND $date_condition");
+    $statistics[4]['percent_parent'] = sprintf("%2.02f", ( $statistics[4]['visits'] / ($statistics[3]['visits']?$statistics[3]['visits']:1) ) * 100);
+    $statistics[4]['percent_all'] = sprintf("%2.02f", ( $statistics[4]['visits'] / ($all_views?$all_views:1) ) * 100);
+    $statistics[5]['step'] = 'final_page';
+    $statistics[5]['visits'] = func_query_first_cell("SELECT count(final_page) FROM $sql_tbl[stats_cart_funnel] WHERE final_page>0 AND $date_condition");
+    $statistics[5]['percent_parent'] = sprintf("%2.02f", ( $statistics[5]['visits'] / ($statistics[4]['visits']?$statistics[4]['visits']:1) ) * 100);
+    $statistics[5]['percent_all'] = sprintf("%2.02f", ( $statistics[5]['visits'] / ($all_views?$all_views:1) ) * 100);
 }
 elseif ($mode == 'cartfunnel') {
 /**
@@ -133,8 +134,9 @@
  */
     $location[] = array(func_get_langvar_by_name('lbl_log_in_history'), '');
 
-    $date = "$sql_tbl[login_history].date_time+'" . $config["Appearance"]["timezone_offset"] . "'";
-    $date_condition = "($date >= '$start_date' AND $date <= '$end_date')";
+    $start_date_off = $start_date - $config["Appearance"]["timezone_offset"];
+    $end_date_off = $end_date - $config["Appearance"]["timezone_offset"];
+    $date_condition = "($sql_tbl[login_history].date_time>='$start_date_off' AND $sql_tbl[login_history].date_time<='$end_date_off')";
 
     if ($REQUEST_METHOD == 'POST') {
 
@@ -151,7 +153,7 @@
         func_header_location("statistics.php?".$QUERY_STRING);
     }
 
-    $statistics = func_query("SELECT $sql_tbl[login_history].*, $sql_tbl[customers].login FROM $sql_tbl[login_history], $sql_tbl[customers] WHERE $sql_tbl[login_history].userid = $sql_tbl[customers].id AND ".$date_condition." ORDER BY date_time DESC");
+    $statistics = func_query("SELECT $sql_tbl[login_history].*,INET_NTOA($sql_tbl[login_history].ip) as s_ip, $sql_tbl[customers].login FROM $sql_tbl[login_history], $sql_tbl[customers] WHERE $sql_tbl[login_history].userid = $sql_tbl[customers].id AND ".$date_condition." ORDER BY date_time DESC");
     if (!empty($statistics)) {
         foreach ($statistics as $k=>$v) {
             $statistics[$k]['date_time'] += $config['Appearance']['timezone_offset'];
