Index: admin/referred_sales.php
--- admin/referred_sales.php	2011-01-10 16:11:47.000000000 +0300
+++ admin/referred_sales.php	2011-01-25 12:43:11.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: referred_sales.php,v 1.22.2.1 2011/01/10 13:11:47 ferz Exp $
+ * @version    $Id: referred_sales.php,v 1.22.2.2 2011/01/25 09:43:11 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -48,7 +48,7 @@
 require './auth.php';
 require $xcart_dir.'/include/security.php';
 
-if(!$active_modules['XAffiliate'])
+if (empty($active_modules['XAffiliate']))
     func_403(23);
 
 $location[] = array(func_get_langvar_by_name('lbl_referred_sales'), '');
