Index: include/ajax_image_manipulations.php
--- include/ajax_image_manipulations.php	2011-01-10 16:11:48.000000000 +0300
+++ include/ajax_image_manipulations.php	2011-03-14 10:41:48.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: ajax_image_manipulations.php,v 1.27.2.1 2011/01/10 13:11:48 ferz Exp $
+ * @version    $Id: ajax_image_manipulations.php,v 1.27.2.2 2011/03/14 07:41:48 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -104,8 +104,8 @@
 
     $id = intval($_GET['id']);
     $type = $_GET['type'];
-    $image_geid = $_GET['image_geid'];
-    $thumbnail_geid = $_GET['thumbnail_geid'];
+    $image_geid = @$_GET['image_geid'];
+    $thumbnail_geid = @$_GET['thumbnail_geid'];
 
     if (!in_array($type, array('P', 'T'))) $type = 'T';
 
