Index: include/checkout.php
--- include/checkout.php	2011-01-10 16:11:48.000000000 +0300
+++ include/checkout.php	2011-02-08 12:10:58.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>. All rights reserved
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: checkout.php,v 1.24.2.2 2011/01/10 13:11:48 ferz Exp $
+ * @version    $Id: checkout.php,v 1.24.2.4 2011/02/08 09:10:58 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -81,7 +81,9 @@
     func_header_location('error_message.php?error_max_items');
 }
 
-$smarty->assign('partner', $partner);
+if (!empty($partner)) {
+    $smarty->assign('partner', $partner);
+}    
 
 if (
     empty($login)
@@ -121,7 +123,9 @@
 }
 
 // Do not show the 'on_registration antibot image' for customers passed verification procedure
-$display_antibot = empty($login) && empty($anonymous_userinfo);
+x_load('user');
+$_anonymous_userinfo = func_get_anonymous_userinfo();
+$display_antibot = empty($login) && empty($_anonymous_userinfo);
 $smarty->assign('display_antibot', $display_antibot);
 
 define('CHECKOUT_STARTED', 1);
