Index: include/common.php
--- include/common.php	2011-01-10 16:11:48.000000000 +0300
+++ include/common.php	2011-01-25 12:43:11.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>. All rights reserved
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: common.php,v 1.9.2.1 2011/01/10 13:11:48 ferz Exp $
+ * @version    $Id: common.php,v 1.9.2.2 2011/01/25 09:43:11 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -92,13 +92,13 @@
             $smarty->assign('categories_menu_list', $categories);
         }
 
-        if ($active_modules['Manufacturers']) {
+        if (!empty($active_modules['Manufacturers'])) {
 
             include $xcart_dir . '/modules/Manufacturers/customer_manufacturers.php';
 
         }
 
-        if ($active_modules['Bestsellers']) {
+        if (!empty($active_modules['Bestsellers'])) {
 
             include $xcart_dir . '/modules/Bestsellers/bestsellers.php';
 
