Index: include/commissions.php
--- include/commissions.php	2011-01-10 16:11:48.000000000 +0300
+++ include/commissions.php	2011-01-20 11:06:35.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>. All rights reserved
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: commissions.php,v 1.10.2.1 2011/01/10 13:11:48 ferz Exp $
+ * @version    $Id: commissions.php,v 1.10.2.3 2011/01/20 08:06:35 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -71,36 +71,40 @@
 
     $where = array();
 
-    if ($search['start_date'] && $search['end_date']) {
+    if (!empty($search['start_date']) && !empty($search['end_date'])) {
         $where[] = $search['end_date']." > $sql_tbl[orders].date AND $sql_tbl[orders].date > ".$search['start_date'];
     }
 
-    if ($search['userid']) {
+    if (!empty($search['userid'])) {
         $where[] = "$sql_tbl[provider_payment].userid = '$search[userid]'";
     }
 
-    if ($search['status']) {
+    if (!empty($search['status'])) {
         $where[] = "$sql_tbl[orders].status = '$search[status]'";
     }
 
-    if ($search['orderid']) {
+    if (!empty($search['orderid'])) {
         $where[] = "$sql_tbl[orders].orderid = '$search[orderid]'";
     }
 
-    if ($search['paid']) {
+    if (!empty($search['paid'])) {
         $where[] = " IF($sql_tbl[provider_commissions].paid = 'Y', 'Y', IF($sql_tbl[orders].status IN ('C','P'), 'A', 'N')) = '$search[paid]'";
     }
 
-    // Exact search by provider (for provider area and $single_mode = false)
-    if (!empty($search_data['commissions']['provider'])) {
-        $search_condition .= " AND $sql_tbl[order].provider='" . $search_data['commissions']['provider'] . "'";
+    if (
+        empty($active_modules['Simple_Mode'])
+        && $current_area == 'P'
+    ) {
+        $where[] =  "$sql_tbl[provider_commissions].userid='$logged_userid'";
+    } elseif (!empty($search['provider'])) {
+        $where[] = " $sql_tbl[provider_commissions].userid='" . $search['provider'] . "'";
     }
 
     if (!empty($where)) {
         $where_condition = " AND " . implode(" AND ", $where);
     }
 
-    $report = func_query("SELECT $sql_tbl[provider_commissions].*, ($sql_tbl[provider_commissions].commissions - $sql_tbl[provider_commissions].paid_commissions) AS not_paid_commissions, $sql_tbl[customers].firstname, $sql_tbl[customers].lastname, $sql_tbl[customers].login, ($sql_tbl[orders].subtotal-$sql_tbl[orders].discount-$sql_tbl[orders].coupon_discount) as subtotal, $sql_tbl[orders].date, $sql_tbl[orders].status AS order_status, IF($sql_tbl[provider_commissions].paid = 'Y', 'Y', IF($sql_tbl[orders].status IN ('C','P'), 'A', 'N')) as paid FROM $sql_tbl[provider_commissions], $sql_tbl[orders], $sql_tbl[customers] WHERE $sql_tbl[provider_commissions].userid = $sql_tbl[customers].id AND $sql_tbl[provider_commissions].orderid = $sql_tbl[orders].orderid AND $sql_tbl[customers].status = 'Y' AND $sql_tbl[customers].usertype = 'P'".$where_condition." ORDER BY $sql_tbl[provider_commissions].add_date, $sql_tbl[customers].id");
+    $report = func_query("SELECT $sql_tbl[provider_commissions].*, ($sql_tbl[provider_commissions].commissions - $sql_tbl[provider_commissions].paid_commissions) AS not_paid_commissions, $sql_tbl[customers].firstname, $sql_tbl[customers].lastname, $sql_tbl[customers].login, ($sql_tbl[orders].subtotal-$sql_tbl[orders].discount-$sql_tbl[orders].coupon_discount) as subtotal, $sql_tbl[orders].date, $sql_tbl[orders].status AS order_status, IF($sql_tbl[provider_commissions].paid = 'Y', 'Y', IF($sql_tbl[orders].status IN ('C','P'), 'A', 'N')) as paid FROM $sql_tbl[provider_commissions] INNER JOIN $sql_tbl[orders] ON $sql_tbl[orders].orderid=$sql_tbl[provider_commissions].orderid INNER JOIN $sql_tbl[customers] ON $sql_tbl[customers].id = $sql_tbl[provider_commissions].userid AND $sql_tbl[customers].status = 'Y' AND $sql_tbl[customers].usertype = 'P' WHERE 1 ".$where_condition." ORDER BY $sql_tbl[provider_commissions].add_date, $sql_tbl[customers].id");
 
     if (!empty($report)) {
 
@@ -114,12 +118,12 @@
 
         $ready = array();
 
-        foreach($users as $v) {
+        foreach($users as $userid) {
 
-            $sum = func_query_first_cell("SELECT SUM($sql_tbl[provider_commissions].commissions) FROM $sql_tbl[provider_commissions], $sql_tbl[orders] WHERE $sql_tbl[provider_commissions].userid = '$v' AND $sql_tbl[provider_commissions].orderid = $sql_tbl[orders].orderid AND $sql_tbl[orders].status IN ('C','P') AND $sql_tbl[provider_commissions].paid != 'Y'");
+            $sum = func_query_first_cell("SELECT SUM($sql_tbl[provider_commissions].commissions) FROM $sql_tbl[provider_commissions], $sql_tbl[orders] WHERE $sql_tbl[provider_commissions].userid = '$userid' AND $sql_tbl[provider_commissions].orderid = $sql_tbl[orders].orderid AND $sql_tbl[orders].status IN ('C','P') AND $sql_tbl[provider_commissions].paid != 'Y'");
 
             if ($sum >= $config['General']['providers_commission_min']) {
-                $ready[$v] = $config['General']['providers_commission_min'];
+                $ready[$userid] = $config['General']['providers_commission_min'];
             }
         }
 
@@ -133,7 +137,7 @@
             $arr = func_query("SELECT firstname, lastname, id AS userid, login FROM $sql_tbl[customers] WHERE id IN ('" . implode("','", array_keys($ready)) . "') ORDER BY login");
             if ($arr) {
                 foreach($arr as $k => $v) {
-                    $arr[$k]['min_paid'] = isset($ready[$v['id']]) ? doubleval($ready[$v['id']]) : 0;
+                    $arr[$k]['min_paid'] = isset($ready[$v['userid']]) ? doubleval($ready[$v['userid']]) : 0;
                 }
                 $smarty->assign('ready', $arr);
             }
@@ -159,12 +163,9 @@
 
 }
 
-$smarty->assign('mode',       $mode);
-$smarty->assign('search',       $search_data['commissions']);
-
-$smarty->assign('start_date', $start_date);
-$smarty->assign('end_date',   $end_date);
-$smarty->assign('main',       'commissions');
+$smarty->assign('mode', $mode);
+$smarty->assign('search', $search_data['commissions']);
+$smarty->assign('main', 'commissions');
 
 if (defined('IS_ADMIN_USER')) {
     $providers = func_query("SELECT id, login, title, firstname, lastname FROM $sql_tbl[customers] WHERE usertype='P' ORDER BY login, lastname, firstname");
