Index: modules/XPayments_Connector/config.php
--- modules/XPayments_Connector/config.php	2011-01-10 16:12:04.000000000 +0300
+++ modules/XPayments_Connector/config.php	2011-04-22 16:14:18.000000000 +0400
@@ -32,23 +32,30 @@
 \*****************************************************************************/
 
 /**
- * Configuration script
+ * Module configuration
  *
  * @category   X-Cart
  * @package    X-Cart
  * @subpackage Modules
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
- * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
+ * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>. All rights reserved
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: config.php,v 1.9.2.1 2011/01/10 13:12:04 ferz Exp $
+ * @version    $Id: config.php,v 1.9.2.2 2011/04/22 12:14:18 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
 
-if ( !defined('XCART_START') ) { header("Location: ../../"); die("Access denied"); }
+if ( !defined('XCART_START') ) { header('Location: ../../'); die('Access denied'); }
 
 if ($config['XPayments_Connector']['xpc_xpayments_url']) {
     $config['XPayments_Connector']['xpc_xpayments_url'] = preg_replace('/\/+$/Ss', '', $config['XPayments_Connector']['xpc_xpayments_url']);
 }
 
+$_module_dir  = $xcart_dir . XC_DS . 'modules' . XC_DS . 'XPayments_Connector';
+/*
+ Load module functions
+*/
+if (!empty($include_func))
+    require_once $_module_dir . XC_DS . 'func.php';
+
 ?>
