Index: modules/XPayments_Connector/xpc_func.php
--- modules/XPayments_Connector/xpc_func.php	2011-01-10 16:12:04.000000000 +0300
+++ modules/XPayments_Connector/xpc_func.php	2011-01-25 12:43:13.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: xpc_func.php,v 1.45.2.11 2011/01/10 13:12:04 ferz Exp $
+ * @version    $Id: xpc_func.php,v 1.45.2.12 2011/01/25 09:43:13 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -604,7 +604,7 @@
 
         $usertypes = array('A');
 
-        if ($active_modules['Simple_Mode']) {
+        if (!empty($active_modules['Simple_Mode'])) {
             $usertypes[] = 'P';
         }
 
