Index: skin/common_files/modules/One_Page_Checkout/profile/address_book_link.tpl
--- skin/common_files/modules/One_Page_Checkout/profile/address_book_link.tpl	2010-11-08 16:55:21.000000000 +0300
+++ skin/common_files/modules/One_Page_Checkout/profile/address_book_link.tpl	2011-02-24 16:50:32.000000000 +0300
@@ -1,25 +1,20 @@
 {*
-$Id: address_book_link.tpl,v 1.2.2.3 2010/11/08 13:55:21 aim Exp $ 
+$Id: address_book_link.tpl,v 1.2.2.4 2011/02/24 13:50:32 aim Exp $ 
 vim: set ts=2 sw=2 sts=2 et:
 *}
 <div class="address-book-link">
-  {if $save_new && $type eq 'B'}
-    <label class="save-new" for="new_B">
-      <input type="checkbox" name="new_address[B]" id="new_B" value="{$addressid}" onclick="javascript: if (this.checked) $('#existing_B').attr('checked', false); $('#new_S').attr('disabled', !this.checked);" />
+  {if $change_mode eq 'Y'}
+    <label class="save-new" for="new_{$type}">
+      <input type="checkbox" name="new_address[{$type}]" id="new_{$type}" value="{$addressid}" onclick="javascript: if (this.checked) $('#existing_{$type}').attr('checked', false);" />
       {$lng.lbl_save_as_new_address}
     </label>
     {if $addressid gt 0}
     <br />
-    <label class="update-existing" for="existing_B">
-      <input type="checkbox" name="existing_address[B]" id="existing_B" value="{$addressid}" onclick="javascript:  if (this.checked) $('#new_B').attr('checked', false); $('#existing_S').attr('disabled', !this.checked); " />
+    <label class="update-existing" for="existing_{$type}">
+      <input type="checkbox" name="existing_address[{$type}]" id="existing_{$type}" value="{$addressid}" onclick="javascript:  if (this.checked) $('#new_{$type}').attr('checked', false); " checked="checked" />
       {$lng.lbl_update_existing_address}
     </label>
     {/if}
-  {elseif $save_new && $type eq 'S'}
-    <input type="hidden" name="new_address[S]" id="new_S" value="{$addressid}" disabled="disabled" />
-    {if $addressid gt 0 && $ship2diff eq 'Y'}
-      <input type="hidden" name="existing_address[S]" id="existing_S" value="{$addressid}" disabled="disabled" />
-    {/if}
   {/if}
   <span class="popup-link">
     <a href="popup_address.php?mode=select&amp;for=cart&amp;type={$type}" onclick="javascript: popupOpen('popup_address.php?mode=select&amp;for=cart&amp;type={$type|escape:"javascript"}'); return false;" title="{$lng.lbl_address_book|escape}">{$lng.lbl_address_book}</a>
