Index: modules/Maintenance_Agent/func.php
--- modules/Maintenance_Agent/func.php	2011-01-10 16:11:59.000000000 +0300
+++ modules/Maintenance_Agent/func.php	2011-02-09 15:32:19.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: func.php,v 1.22.2.1 2011/01/10 13:11:59 ferz Exp $
+ * @version    $Id: func.php,v 1.22.2.3 2011/02/09 12:32:19 ferz Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -68,7 +68,7 @@
 {
     global $config;
     global $sql_tbl;
-    global $mail_smarty;
+    global $mail_smarty, $active_modules;
 
     x_load('mail');
 
@@ -119,7 +119,7 @@
 
     if ($config['Maintenance_Agent']['periodic_visits'] == 'Y') {
         // collect informaition about logins
-        $visits = func_query("SELECT userid, usertype, COUNT(*) AS visits, max(date_time) as date_time FROM $sql_tbl[login_history] WHERE date_time>='$date_from' AND date_time<='$date_to' AND action='login' AND usertype IN ('A','P') GROUP BY login, usertype ORDER BY usertype ASC, visits DESC, login ASC");
+        $visits = func_query("SELECT userid, usertype, COUNT(*) AS visits, max(date_time) as date_time FROM $sql_tbl[login_history] WHERE date_time>='$date_from' AND date_time<='$date_to' AND action='login' AND usertype IN ('A','P') GROUP BY userid, usertype ORDER BY usertype ASC, visits DESC, userid ASC");
         if (!is_array($visits)) $visits = array();
 
         func_get_usertype_generic_stat($visits, 'C', $date_from, $date_to);
