Index: skin/common_files/admin/main/register.tpl
--- skin/common_files/admin/main/register.tpl	2010-08-04 10:54:14.000000000 +0400
+++ skin/common_files/admin/main/register.tpl	2011-03-03 13:53:08.000000000 +0300
@@ -1,5 +1,5 @@
 {*
-$Id: register.tpl,v 1.4 2010/08/04 06:54:14 igoryan Exp $
+$Id: register.tpl,v 1.4.2.1 2011/03/03 10:53:08 aim Exp $
 vim: set ts=2 sw=2 sts=2 et:
 *}
 {include file="check_email_script.tpl"}
@@ -76,7 +76,7 @@
   <br />
 {/if}
 
-<form action="{$register_script_name}?{$smarty.server.QUERY_STRING|amp}" method="post" name="registerform" onsubmit="javascript: return checkRegFormFields(this);">
+<form action="{$register_script_name}?{$smarty.server.QUERY_STRING|amp}" method="post" name="registerform" onsubmit="javascript: return checkRegFormFields(this);" {if $config.User_Profiles.skip_js_validation_admin eq "Y"}class="skip-auto-validation"{/if}>
 
 {if $config.Security.use_https_login eq "Y"}
   <input type="hidden" name="{$XCARTSESSNAME}" value="{$XCARTSESSID}" />
