Index: modules/Recently_Viewed/config.php
--- modules/Recently_Viewed/config.php	2011-01-10 16:12:01.000000000 +0300
+++ modules/Recently_Viewed/config.php	2011-04-22 16:14:17.000000000 +0400
@@ -34,18 +34,26 @@
 /**
  * Module configuration
  *
- * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
- * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
- * @license    http://www.x-cart.com/license.php X-Cart license agreement
  * @category   X-Cart
- * @package    Modules
+ * @package    X-Cart
  * @subpackage Recently Viewed
- * @version    $Id: config.php,v 1.4.2.1 2011/01/10 13:12:01 ferz Exp $
+ * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
+ * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>. All rights reserved
+ * @license    http://www.x-cart.com/license.php X-Cart license agreement
+ * @version    $Id: config.php,v 1.4.2.2 2011/04/22 12:14:17 aim Exp $
  * @since      4.4.0
+ * @see        ____file_see____
  */
 
-if (!defined('XCART_START')) { header('Location: ../../'); die('Access denied'); }
+if ( !defined('XCART_START') ) { header('Location: ../../'); die('Access denied'); }
 
 $css_files['Recently_Viewed'][] = array();
 
+$_module_dir  = $xcart_dir . XC_DS . 'modules' . XC_DS . 'Recently_Viewed';
+/*
+ Load module functions
+*/
+if (!empty($include_func))
+    require_once $_module_dir . XC_DS . 'func.php';
+
 ?>
