Index: include/payment/func.cc_usaepay.php
--- include/payment/func.cc_usaepay.php	2011-01-10 16:11:53.000000000 +0300
+++ include/payment/func.cc_usaepay.php	2011-02-07 18:34:45.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: func.cc_usaepay.php,v 1.19.2.2 2011/01/10 13:11:53 ferz Exp $
+ * @version    $Id: func.cc_usaepay.php,v 1.19.2.3 2011/02/07 15:34:45 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -112,6 +112,8 @@
  */
 function func_cc_usaepay_do_sale($bill_output, $authonly, $secure_oid, $userinfo, $products, $giftcerts, $total_cost, $cmpi_result = null)
 {
+    global $bill_name;
+
     $module_params = func_get_pm_params('cc_usaepay.php');
 
     $post = array(
