Index: payment/ps_paypal_pro_uk.php
--- payment/ps_paypal_pro_uk.php	2011-01-10 16:12:08.000000000 +0300
+++ payment/ps_paypal_pro_uk.php	2011-02-24 16:50:32.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: ps_paypal_pro_uk.php,v 1.33.2.4 2011/01/10 13:12:08 ferz Exp $
+ * @version    $Id: ps_paypal_pro_uk.php,v 1.33.2.6 2011/02/24 13:50:32 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -131,14 +131,14 @@
     x_session_register('login_type');
     x_session_register('logged_userid');
 
+    x_load('user');
     if (!empty($login) && $login_type == 'C') {
         
-        $cart['used_s_address'] = $address;
+        $cart = func_set_cart_address($cart, 'S', $address);
 
     }
     elseif ($config['General']['enable_anonymous_checkout'] == 'Y') {
 
-        x_load('crypt','user');
 
         // Fill-in anonymous customer profile
 
@@ -166,10 +166,9 @@
             $pp_anon_user['address']['S'][$k] = $v;
         }
 
+        x_load('crypt');
         // save anonymous customer info in session
-        x_session_register('anonymous_userinfo', array());
-        $anonymous_userinfo = $pp_anon_user;
-        $anonymous_userinfo['usertype'] = empty($usertype) ? 'C' : $usertype;
+        func_set_anonymous_userinfo($pp_anon_user);
 
     }
     else {
