Index: modules/Google_Checkout/init.php
--- modules/Google_Checkout/init.php	2011-01-10 16:11:58.000000000 +0300
+++ modules/Google_Checkout/init.php	2011-02-04 19:54:14.000000000 +0300
@@ -40,7 +40,7 @@
  * @author     Ruslan R. Fazlyev <rrf@x-cart.com>
  * @copyright  Copyright (c) 2001-2011 Ruslan R. Fazlyev <rrf@x-cart.com>
  * @license    http://www.x-cart.com/license.php X-Cart license agreement
- * @version    $Id: init.php,v 1.18.2.1 2011/01/10 13:11:58 ferz Exp $
+ * @version    $Id: init.php,v 1.18.2.2 2011/02/04 16:54:14 aim Exp $
  * @link       http://www.x-cart.com/
  * @see        ____file_see____
  */
@@ -109,7 +109,7 @@
     register_shutdown_function('func_gcheckout_save_log');
 }
 
-if (AREA_TYPE != 'C')
+if (func_constant('AREA_TYPE') != 'C')
     return;
 
 if (!$gcheckout_enabled) {
