Index: skin/common_files/modules/Survey/modify_maillist.tpl
--- skin/common_files/modules/Survey/modify_maillist.tpl	2010-06-15 11:58:18.000000000 +0400
+++ skin/common_files/modules/Survey/modify_maillist.tpl	2011-01-20 11:03:46.000000000 +0300
@@ -1,5 +1,5 @@
 {*
-$Id: modify_maillist.tpl,v 1.4 2010/06/15 07:58:18 igoryan Exp $
+$Id: modify_maillist.tpl,v 1.4.2.1 2011/01/20 08:03:46 aim Exp $
 vim: set ts=2 sw=2 sts=2 et:
 *}
 {include file="main/multirow.tpl"}
@@ -47,7 +47,7 @@
   {if $config.email_as_login ne 'Y'}
   <td nowrap="nowrap">{$m.email}</td>
   {/if}
-  <td nowrap="nowrap" align="center">{if $m.login}<a href="user_modify.php?user={$m.userid}&amp;usertype=C">{$m.login}</a>{else}{$lng.txt_not_available}{/if}</td>
+  <td nowrap="nowrap" align="center">{if $m.login}<a href="{$catalogs.admin}/user_modify.php?user={$m.userid}&amp;usertype=C">{$m.login}</a>{else}{$lng.txt_not_available}{/if}</td>
   <td align="center" nowrap="nowrap">
 {if $m.complete_date}
 {assign var="date" value=$m.complete_date|date_format:$config.Appearance.datetime_format}
